box/src/main/java/ch/cyberduck/core/box/BoxWriteFeature.java

```diff
@@ -69,32 +79,30 @@ public BoxWriteFeature(final BoxSession session, final BoxFileidProvider fileid)
                     final HttpPost request;
                     if(status.isExists()) {
                         request = new HttpPost(String.format("%s/files/%s/content?fields=%s", client.getBasePath(),
-                                fileid.getFileId(file, new DisabledListProgressListener()),
-                                String.join(",", BoxAttributesFinderFeature.DEFAULT_FIELDS)));
+                            fileid.getFileId(file, new DisabledListProgressListener()),
+                            String.join(",", BoxAttributesFinderFeature.DEFAULT_FIELDS)));
                     }
                     else {
                         request = new HttpPost(String.format("%s/files/content?fields=%s", client.getBasePath(),
-                                String.join(",", BoxAttributesFinderFeature.DEFAULT_FIELDS)));
+                            String.join(",", BoxAttributesFinderFeature.DEFAULT_FIELDS)));
                     }
                     final Checksum checksum = status.getChecksum();
                     if(Checksum.NONE != checksum) {
-                        switch(checksum.algorithm) {
-                            case sha1:
-                                request.addHeader(HttpHeaders.CONTENT_MD5, checksum.hash);
+                        if(checksum.algorithm == HashAlgorithm.sha1) {
+                            request.addHeader(HttpHeaders.CONTENT_MD5, checksum.hash);
                         }
                     }
-                    final ByteArrayOutputStream content = new ByteArrayOutputStream();
-                    new JSON().getContext(null).writeValue(content, new FilescontentAttributes()
-                            .name(file.getName())
-                            .parent(new FilescontentAttributesParent().id(fileid.getFileId(file.getParent(), new DisabledListProgressListener())))
-                            .contentModifiedAt(status.getTimestamp() != null ? new DateTime(status.getTimestamp()) : null)
-                    );
+                    final FilescontentAttributes filescontentAttributes = new FilescontentAttributes()
+                        .name(file.getName())
+                        .parent(new FilescontentAttributesParent().id(fileid.getFileId(file.getParent(), new DisabledListProgressListener())))
+                        .contentModifiedAt(status.getTimestamp() != null ? new DateTime(status.getTimestamp()) : null);
+                    final byte[] content = getObjectMapper().writer().writeValueAsBytes(filescontentAttributes);
```

Comment:
We use `new JSON().getContext(null)` everywhere else to obtain the mapper. Why not use this and add the custom date time serializer module there?

Reply:
If I use that, then it includes the registered Joda Module. I do not find any option to unregister that registered module.